### PR TITLE
HDDS-9548. Intermittent failures in TestRootCaRotationPoller

### PR DESCRIPTION
**Pull request by Galsza**
## What changes were proposed in this pull request?
Fix the failure in TestRootCaRotationPoller arising from multi threaded execution

Please describe your PR in detail:
Sometimes the test cases were failing because `CompletableFuture.supplyAsync()` wasn't finishing before assertions were made for its results. Now 2 test cases avoid using the future at all, the third one is refactored so that it can wait for its result.

There is also a small refactor regarding pollRootCas, its visibility is changed so that it can be called directly from the test instead of relying on the run method that would introduce another point of possible sync issue via the executorService scheduling.

And finally pollRootCas now returns weather it found new certificates. Also added some logging in the success scenario.

The error `Exception in thread "RootCaRotationPoller" java.lang.OutOfMemoryError: GC overhead limit exceeded` mentioned in the Jira ticked was unrelated, its was caused by my local environment. (IntelliJ tried storing all test results on the heap)

## How was this patch tested?

Run each test locally 1000 times. I will add additional CI runs

Benchmark CI run on master finding at least 2 failures: https://github.com/Galsza/ozone/actions/runs/6847473750/job/18618436190
Run on the current branch without failures: 
https://github.com/Galsza/ozone/actions/runs/6849624033/job/18622199979

### CODE REVIEW

**Comment by Galsza** [comment]
@adoroszlai and @ChenSammi please review.

This is only in draft state right now because I want to add CI runs where I reproduce the original failure ratio without the change, and runs with the new change to see if there are no failures any more.

File: hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/x509/certificate/client/TestRootCaRotationPoller.java
Comment (ChenSammi) [inline hadoop-hdds/framework/src/test/java/org/apache/hadoop/hdds/security/x509/certificate/client/TestRootCaRotationPoller.java:85]: If we just check the poller.pollRootCas() status, then this test doesn't match the test function name, as the invoke is not checked.  The way used in testPollerRetriesAfterFailure() can be used here too.

**Comment by Galsza** [comment]
@adoroszlai @ChenSammi Please take a look I've updated the pr and added CI runs testing the changes.

**Review by adoroszlai — COMMENTED** [review]
Thanks @Galsza for updating the patch, LGTM.  I'll let @ChenSammi take another look before merging it.

**Review by ChenSammi — APPROVED** [review]
The final patch looks good to me.

**Comment by adoroszlai** [comment]
Thanks @ChenSammi for the review.

**Comment by Galsza** [comment]
Thanks @adoroszlai and @ChenSammi for the reviews!

**Pull request merged**
